### ISSUE TITLE
registering a submodule dynamically

### ISSUE THREAD

**Issue by ReinisV** [sascha245/vuex-simple#23]
-  **I'm submitting a ...**
[ ] bug report
[ ] feature request
[ ] question about the decisions made in the repository
[x] question about how to use this project

-  **Summary**
how does one register a module that is defined using typescript classes dynamically?
https://vuex.vuejs.org/guide/modules.html#dynamic-module-registration
This is specifically for use with JS Vue components, that have not yet been converted to `vue-class-component` TS components.
*  **Other information** (e.g. detailed explanation, stacktraces, related issues, suggestions how to fix, links for us to have context, eg. StackOverflow, personal fork, etc.)

importing the class based submodule and registering it to the store using the `registerModule` method does not seem to work. 
```
import { SomeVuexModule } from '@store/modules/some-module';
beforeCreate() {
this.$store.registerModule('someModule', SomeVuexModule);
}
```
or
```
import { SomeVuexModule } from '@store/modules/some-module';
beforeCreate() {
this.$store.registerModule('someModule', new SomeVuexModule());
}
```
I think that's because `registerModule` requires a raw (already mapped) module, but I do not know how to get one from the class based submodule.

**Comment by sascha245** [comment]
Hi @ReinisV,

Dynamical module registration isn't supported for now. However, it wouldn't be that difficult to implement, as there already is an internal function that converts an instance to a vuex module.

I added this feature to my todo list and will write a comment here once it is done.

**Comment by ReinisV** [comment]
Is the function your talking about the [transformRecursive](https://github.com/sascha245/vuex-simple/blob/b0121efe03228523ed912519305f3f34351c49d3/src/store.ts#L57)?

It'd be awesome if I could figure out a way to call it directly for my use case.

**Comment by sascha245** [comment]
Yep it is this function I was talking about, you can probably write a function that wraps a call to transformRecursive to do what you want:
```ts
function createVuexModule(store, instance, namespace: string): Module<any, any> {
  const storeProvider: StoreProvider = { store };
  return transformRecursive(storeProvider, instance, namespace.split('/'));
}
```
This should transform the instance and return the module config, though I just did that on a fly and didn't test it.

**Comment by sascha245** [comment]
Dynamic modules have been officially added to the package. There is a section in the README going over the 3 functions that have been implemented in #25:
https://github.com/sascha245/vuex-simple#dynamic-modules

Though all the tests passed and everything in the samples worked, don't hesitate to signal any bugs that may have appeared in this version.
